src/book/04-general/01-git.mdx

```diff
@@ -0,0 +1,59 @@
+---
+section: Guides
+chapter: General
+title: Git
+description: Contributing to a NUbots project with git.
+slug: /guides/general/git
+---
+
+## Prerequisites
+
+This guide assumes you know the basics of git and basic shell commands.
+
+## Getting Code Into a Repository
+
+Suppose you have a file you have made which you would like to add to a NUbots repository, called `my_file.cpp`, as part of your project to make the robot stand up.
+
+In the terminal, change directory to where you have the repo cloned.
+
+### Select Branch
+
+If you are working within an existing project, then you need to find out which branch the work is on. To list the branches, use `git branch`. When you have found the branch you want to push to, swap to it:
+
+```sh
+git checkout <branch name>
+```
+
+If you are not working on an existing project, you need to make a branch to commit your changes to. The format for branch names is `surname/project_name`. So for this example it could be `citizen/stand_up`.
```

Comment:
```suggestion
Where `<branch name>` corresponds to the branch you found in the last step. 

If you are not working on an existing project, you need to make a branch to commit your changes to. The format for branch names is `surname/project_name`. So for this example it could be `citizen/stand_up`, if your surname is Citizen.
```